calcs/depthBelowKeel2.js

```diff
@@ -1,28 +1,37 @@
 const _ = require('lodash')
 
 module.exports = function (app) {
-  var depthTransducerToKeel = app.getSelfPath(
-    'environment.depth.transducerToKeel.value'
-  )
-
-  var derivedFrom =
-    typeof depthTransducerToKeel === 'undefined'
-      ? []
-      : ['environment.depth.belowTransducer']
+  let depthTransducerToKeel = app.getSelfPath('environment.depth.transducerToKeel.value') || 0
 
   return {
     group: 'depth',
     optionKey: 'belowKeel_2',
-    title:
-      'Depth Below Keel (based on depth.belowTransducer and depth.transducerToKeel)',
-    derivedFrom: derivedFrom,
-    calculator: function (depthBelowTransducer) {
+    title: 'Depth Below Keel (based on depth.belowTransducer and depth.transducerToKeel)',
+    derivedFrom: ['environment.depth.belowTransducer', 'environment.depth.transducerToKeel'],
+    calculator: function (depthBelowTransducer, transducerToKeel) {
```

Comment:
This will never get called if there are no values in the `environment.depth.transducerToKeel`stream and will fire an extra depth message every time there is a value in that stream, even if the value has not changed.

I would just a separate subscription to that stream via streambundle and update the original depthTransducerToKeel variable. This way only depthBelowTransducer values result in output and the calculation always uses the latest depthTransducerToKeel value.

Reply:
@tkurki thanks, can I do that using `app.getSelfPath` int he calculator function?

Reply:
@tkurki like so?

Reply:
Looks good to me.